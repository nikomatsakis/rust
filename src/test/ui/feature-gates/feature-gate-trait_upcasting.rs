trait Foo {}

trait Bar: Foo {}

impl Foo for () {}

impl Bar for () {}

fn main() {
    let bar: &dyn Bar = &();
    let foo: &dyn Foo = bar;
    //~^ ERROR mismatched types [E0308]
    //~| NOTE expected trait `Foo`, found trait `Bar`
    //~| NOTE expected due to this
    //~| NOTE expected reference `&dyn Foo`
    //~| NOTE add `#![feature(trait_upcasting)]` to the crate attributes to enable trait upcasting
}
