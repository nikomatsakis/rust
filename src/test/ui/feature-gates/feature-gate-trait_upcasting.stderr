error[E0308]: mismatched types
  --> $DIR/feature-gate-trait_upcasting.rs:11:25
   |
LL |     let foo: &dyn Foo = bar;
   |              --------   ^^^ expected trait `Foo`, found trait `Bar`
   |              |
   |              expected due to this
   |
   = note: expected reference `&dyn Foo`
              found reference `&dyn Bar`
   = note: add `#![feature(trait_upcasting)]` to the crate attributes to enable trait upcasting

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
